src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/bqsr/BaseRecalibratorFn.java

```diff
@@ -333,11 +259,22 @@ protected boolean badSolidOffset(final GATKRead read, final int offset) {
         return knownSites;
     }
 
+
+    private boolean isLowQualityBase(final GATKRead read, final int offset) {
+        return read.getBaseQualities()[offset] < minimumQToUse;
+    }
+
+    private boolean badSolidOffset(final GATKRead read, final int offset) {
+        return ReadUtils.isSOLiDRead(read, header) && BRAC.RAC.SOLID_RECAL_MODE != RecalUtils.SOLID_RECAL_MODE.DO_NOTHING && !RecalUtils.isColorSpaceConsistent(read, offset);
+    }
+
     // TODO: can be merged with calculateIsIndel
-    protected static int[] calculateIsSNP(final GATKRead read, /*final BasesCache ref*/ final ReferenceContext ref, final GATKRead originalRead) {
+    private static int[] calculateIsSNP(final GATKRead read, final ReferenceContext ref, final GATKRead originalRead) {
         final byte[] readBases = read.getBases();
         final byte[] refBases = Arrays.copyOfRange(ref.getBases(), read.getStart() - originalRead.getStart(), ref.getBases().length + read.getEnd() - originalRead.getEnd());
```

Comment:
I don't think that this statement will produce an array with the correct reference bases if `read.getStart() != originalRead.getStart()` or `read.getEnd() != originalRead.getEnd()`. The previous assumption was that `ref.getBases()` contained exactly the bases that span the `originalRead`, which is no longer the case.